src/CreateInvalidArgumentExceptionCapableTrait.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Dhii\Exception;
+
+use Exception as RootException;
+use InvalidArgumentException as RootInvalidArgumentException;
+
+trait CreateInvalidArgumentExceptionCapableTrait
+{
+    /**
+     * Creates a new invalid argument exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null $message  The error message, if any.
+     * @param int|null               $code     The error code, if any.
+     * @param RootException|null     $previous The inner exception for chaining, if any.
+     * @param mixed|null             $argument The invalid argument, if any.
+     *
+     * @return InvalidArgumentException The new exception.
+     */
+    protected function _createInvalidArgumentException(
```

Comment:
While I agree with **why** this was introduced in this package, I can't help but feel like _can_ result in re-declaration errors, given that it already exists (same name and all) in another package.

https://github.com/Dhii/exception-helper-base/blob/develop/src/CreateInvalidArgumentExceptionCapableTrait.php

Notice that they are identical. The only differences are the doc block and the alias for `InvalidArgumentException`.

Even without autoloading, this presents some risks. How can we avoid this?

Reply:
Excellent catch. I've renamed it now to avoid the clash.

On a related note, this really brings out the benefits of a possible class indexing tool.

Reply:
By the way, @mecha, the helper factory creates a `Dhii\Exception\InvalidArgumentException`, not a native one, while the trait in this package creates a native PHP exception. That's the difference.

Reply:
Yeah 👍 

At least ... now it does 😉

Reply:
Always did. Last commit was made 7 days ago, and was a Composer config change ;)